src/backend/distributed/transaction/multi_shard_transaction.c

```diff
@@ -69,9 +72,24 @@ OpenTransactionsToAllShardPlacements(List *shardIntervalList, int connectionFlag
 								   UINT64_FORMAT, shardId)));
 		}
 
+		if (task->taskType == MODIFY_TASK)
+		{
+			accessType = PLACEMENT_ACCESS_DML;
+		}
+		else
+		{
+			/* can only open connections for DDL and DML commands */
+			Assert(task->taskType == DDL_TASK);
+
+			accessType = PLACEMENT_ACCESS_DDL;
+		}
+
 		foreach(placementCell, shardPlacementList)
 		{
 			ShardPlacement *shardPlacement = (ShardPlacement *) lfirst(placementCell);
+			ShardPlacementAccess placementModification;
```

Comment:
In this case it's easy to allocate on the stack, which is always simpler and more efficient.